WordPress/gutenberg
# ProgressBar: Add default `tabIndex` to make focusable

## What?
This PR updates the `ProgressBar` component to make it focusable.

cc @alexstine @andrewhayward for accessibility feedback.

## Why?
While working on #53032 I noticed that the underlying `progress` element is not focusable and that makes it inaccessible. That meant we have to make it focusable so users with screen readers will be able to focus it and get it announced properly.

## How?
This PR adds a default `tabIndex` of `0` to the underlying `progress` HTML element to make it focusable. We still allow the component user to override the value if they wish.

I'm also adding a unit test to verify the focusing works as expected. You can try removing the `tabIndex` and you should see the new test failing.

## Testing Instructions
* This PR is part of #53032, so you can test it there.
* Alternatively, spin up Storybook with `npm run storybook:dev`, load http://localhost:50240/?path=/story/components-experimental-progressbar--default
* Verify that the progress bar is focusable and announces properly that it's a progress bar in a loading state with a screen reader.
* Verify all tests pass - the change is covered by a new test I'm introducing. 

### Testing Instructions for Keyboard
Whether in the site editor or in Storybook, the element should be the sole focusable element in the canvas, so tabbing once should be enough.

## Screenshots or screencast 
None.
